src/test/java/bio/terra/service/upgrade/MigrateTests.java

```diff
@@ -0,0 +1,172 @@
+package bio.terra.service.upgrade;
+
+import bio.terra.app.configuration.ApplicationConfiguration;
+import bio.terra.app.configuration.StairwayJdbcConfiguration;
+import bio.terra.app.logging.PerformanceLogger;
+import bio.terra.common.category.Unit;
+import bio.terra.service.iam.IamService;
+import bio.terra.service.job.JobService;
+import bio.terra.service.resourcemanagement.google.GoogleResourceConfiguration;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.ApplicationContext;
+import org.springframework.core.env.Environment;
+import org.springframework.core.env.StandardEnvironment;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.util.Collections;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+
+@RunWith(SpringRunner.class)
+@Category(Unit.class)
+@SpringBootTest
+@ActiveProfiles({"google", "test"})
+@AutoConfigureMockMvc
+public class MigrateTests {
+
+    @Autowired
+    private IamService samService;
```

Comment:
It looks like these are mocked below, if so they don't need to be Autowired here, they can be created in the constructor call to `JobService`.

Or, you could use `@MockBean` and also override the autowiring for `MigrateConfiguration`, `Environment` and `GoogleResourceConfiguration`. If you did that, then you could `Autowire` `JobService` and it would inject the mocks.

Reply:
I've removed the autowire!